### PR TITLE
ISSUE: 1585 | Add --wait flag to databases resize command

### PR DESCRIPTION
**Pull request by Yordaniss**
ISSUE #1585 
* added bool flag into resize command
* added wait functionality into resize
* adjusted test

### CODE REVIEW

File: commands/databases.go [hunk 1]
```diff
@@ -511,13 +513,57 @@ func RunDatabaseResize(c *CmdConfig) error {
...
+	err = dbs.Resize(id, r)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve the database object after resize (to check status or further actions)
+	db, err := dbs.Get(id)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve the database after resize: %v", err)
+	}
```
Comment (andrewsomething) [inline commands/databases.go:533]: I think that we could probably simplify this a bit further. Compared to the create command, the resize command only returns success or error. It doesn't output the database information. So we can remove this call to `Get` here as well as the one below in the `wait` block.

File: commands/databases.go [hunk 2]
```diff
@@ -511,13 +513,57 @@ func RunDatabaseResize(c *CmdConfig) error {
...
+				"failed to retrieve the resized database: %v",
+				err,
+			)
+		}
+		db.Connection = connection
+	}
+
+	notice("Database resized successfully")
+
+	return displayDatabases(c, false, *db)
```
Comment (andrewsomething) [inline commands/databases.go:566]:
```suggestion
		notice("Database resized successfully")
	}

	return nil
```

Let's also move the notice inside the wait block. That way the current behavior remains the same if you do not pass the `--wait` flag.

**Review by andrewsomething — CHANGES_REQUESTED** [review]
Hi @Yordaniss. Thank you so much for the pull request!

I've left a few comments in-line with suggestions. In addition to that, it would be great to add a new test case to the `TestDatabaseResize` test. 
https://github.com/digitalocean/doctl/blob/6804c7d62dfd7ff142ddb3ac1abed3893f9cdcab/commands/databases_test.go#L618

Something along the line of:

```go
	// Success with wait flag
	withTestClient(t, func(config *CmdConfig, tm *tcMocks) {
		tm.databases.EXPECT().Resize(testDBCluster.ID, r).Return(nil)
		tm.databases.EXPECT().Get(testDBCluster.ID).Return(&testDBCluster, nil)
		config.Args = append(config.Args, testDBCluster.ID)
		config.Doit.Set(config.NS, doctl.ArgSizeSlug, testDBCluster.SizeSlug)
		config.Doit.Set(config.NS, doctl.ArgDatabaseNumNodes, testDBCluster.NumNodes)
		config.Doit.Set(config.NS, doctl.ArgDatabaseStorageSizeMib, testDBCluster.StorageSizeMib)
		config.Doit.Set(config.NS, doctl.ArgCommandWait, true)

		err := RunDatabaseResize(config)
		assert.NoError(t, err)
	})
```

Comparing that to the existing tests, you'll see it passes the `wait` flag and tests that the `Get` method is called by `waitForDatabaseReady` when the flag is passed.

Let us know if you have any questions!

**Comment by Yordaniss** [comment]
Hi @andrewsomething thanks a lot for a feedback. I simplified logic and added test case. 🙂

**Review by andrewsomething — APPROVED** [review]
:+1:  Great work! Thanks again for the contribution!

**Pull request merged**